universum/main.py

```diff
@@ -88,7 +90,9 @@ def execute(self) -> None:
                     self.code_report_collector.repo_diff = repo_diff
             self.code_report_collector.report_code_report_results()
         self.artifacts.report_artifacts()
-        self.reporter.report_build_result()
+        result = self.reporter.report_build_result()
+        if self.settings.fail_unsuccessful and not result:
```

Comment:
Does this condition rely on "block" context manager with pass_error=False returning None (not returning anything) in case of exception?
If that is the case, I would say this implicit dependency is far from obvious.